### ISSUE TITLE
Feature Request: faster zotero searcher with better search results for creating/opening zotero notes

### ISSUE THREAD

**Issue by whuzjw** [mgmeyers/obsidian-zotero-integration#103]
Hi,

Thanks so much for developing this useful plugin. This plugin can open a zotero searcher for creating notes like below:
![image](https://user-images.githubusercontent.com/54569058/194541244-25076f95-0d5c-4e19-a280-8ac4e507e925.png)

However, there are two main **problems**:
1. it is **slow** and sometimes the window of zotero searcher may **not display** (only display the window of "Awaiting" like Fig.1), thus user has to manually switch to the zotero searcher windows by ALT+TAB (Fig.2).  
Fig.1 ![image](https://user-images.githubusercontent.com/54569058/194542752-9ccea9a3-4b31-48d3-94fb-f64dfe72c91b.png)
Fig.2 ![image](https://user-images.githubusercontent.com/54569058/194542847-1f607912-8071-4c88-b930-a9240e3ba47c.png)
2. The searching results are **not well ranked**. E.g., in the figure below, the second record should ranked first. 
![image](https://user-images.githubusercontent.com/54569058/194543301-2b79f21e-4ad7-4f7a-92ae-60b95d4fab7c.png)

**Recently**, there is a plugin called [obsidian-zotero](https://github.com/aidenlx/obsidian-zotero). It opens a built-in obsidian searcher called **"quick switcher"** for creating/opening zotero notes, which is fast (Fig.3) and gives better accuracy (Fig.4).
Fig.3 ![image](https://user-images.githubusercontent.com/54569058/194540912-cc7cdc0c-3872-4747-b75d-e61913669871.png)
Fig.4 ![image](https://user-images.githubusercontent.com/54569058/194543515-bd90458c-0ccd-48f9-9f14-63351e81e51f.png)

**However**, when I searched  StyleGAN2 by its author name, it failed (Fig.5 6). 
Fig.5 ![image](https://user-images.githubusercontent.com/54569058/194545644-8552b6a9-c040-4d3e-9950-16072d6011c8.png)
Fig.6 ![image](https://user-images.githubusercontent.com/54569058/194545711-a51d1aa2-1468-4dda-ac55-9a3608248f13.png)

**So**, could you implement a "quick switcher" with a scope including the author's name?

Thank you again!

**Comment by ryanwwest** [comment]
I agree that it's pretty slow and not always reliable which is why I continue to use https://github.com/hans/obsidian-citation-plugin (even though this may have more features). That plugin can be fast because it uses an exported citations library in json format (that Zotero automatically keeps updated), whereas I think this directly interfaces with the Zotero app, which might cause more latency.

I hadn't heard of https://github.com/aidenlx/obsidian-zotero, thanks for pointing it out.

**Comment by MSBack** [comment]
I second the call for speed improvements. This would be a major upgrade to Zotero Integration workflow, without the friction of waiting so long for each literature note to import. Having used Bibnotes and Citations, the speed difference is particularly glaring. Of course, those plugins have their own bugs and shortcomings. But I think this speed issue lies mainly on Zotero's side, as the native citation picker is slow in Word and elsewhere too.

The best way to improve the speed without relying on exported bibliography files would probably be to adopt the backend of Obsidian Zotero, as mentioned, but that implies a major rewrite.

**Comment by lseebach** [comment]
+1
I wonder if so could get the best of two worlds:

1. the quick and obsidian-native lookup from the [citation citation plugin](https://github.com/hans/obsidian-citation-plugin) (I suppose the lookup is based on an "offline" representation of the Zotero repository instead of using an Zotero API call for each lookup) and 
2. the extensive amount of prpereties and flexibility in templating of this plugin

**Comment by danieltomasz** [comment]
Maybe this could be split into 2 issues
- implementing native (in-obsidina) literature picker
- Improving its capabilities

**Comment by MSBack** [comment]
> Maybe this could be split into 2 issues
> 
> - implementing native (in-obsidina) literature picker
> 
> - Improving its capabilities

Well, it's hard to know if such a native picker would need improvement beyond its hypothetical initial implementation. But I still think going the route of the Obsidian-Zotero plugin by @aidenlx, with direct access to the Zotero database, would be the best option. Obsidian-Zotero also provides an API now, but I guess that would mean having both plugins installed, which seems a bit redundant.

I know the Bibnotes formatter maintainer is investigation the use of the Obsidian-Zotero API, as he requested for it to be implemented.
